test/reload_system.bats

```diff
@@ -25,8 +37,24 @@ function expect_log_failure() {
 	wait_for_log "system registries reload failed"
 }
 
+function count_log_entries() {
+	local search_pattern="$1"
+	local count
+
+	# Use grep to count occurrences of the pattern.
+	count=$(grep -ico "$search_pattern" "$CRIO_LOG")
+
+	# Check if grep found any occurrences, if not, return 0.
+	if [[ -z "$count" ]]; then
+		echo 0
+	else
+		echo "$count"
+	fi
```

Comment:
At least on Linux using GNU version of grep, with the `-c` and `-o` options, it would return 0 and n, per:

```console
$ t() { grep -co "$1" "$2" ; } 
$ t "something" /etc/resolv.conf
0
$ t "nameserver" /etc/resolv.conf
1
```

As such, you could simply this function a little bit.